Donation-platform/src/main/webapp/personalizationScript.js

```diff
@@ -107,9 +135,203 @@ function displayTags(tags) {
 $(function() {
   $("#sortable").sortable();
   $('#submit').click(function() {
+
+    // Remove the first element in tagOrder (which is the instruction label)
+    // in order to get the tags.
     var tagOrder = $("#sortable").sortable('toArray');
-    console.log(tagOrder);
-    loadPersonalizedCharities(tagOrder);
+    var tags = [];
+    for(i = 1; i < tagOrder.length; i++) {
+      tags.push(tagOrder[i]);
+    }
+
+    // If there are 3 tags in the rank, run loadPersonalizedCharities.
+    if(tags.length == 3) {
+      loadPersonalizedCharities(tags);
+    // Otherwise, alert the user to select 3 causes before re-submitting.
```

Comment:
we should not force user to select atleast 3. It should be maximum of 3. Can we have "2 nd part " as dynamic to show only 1/2/3 ? if not then we should choose other 2 randomly.

Reply:
Done.